### PR TITLE
internal/v5: allow POST with ingest flag

### PR DESCRIPTION
**Pull request by rogpeppe**
Allow POST with an "ingest" parameter to indicate that although it's a
POST, it shouldn't turn off ingestion. This will enable us to run non-legacy
ingestion which uses the POST method without disabling itself.

### CODE REVIEW

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/charmstore/1744/
Test PASSed.

File: internal/v5/archive_test.go [hunk 1]
```diff
@@ -672,6 +672,39 @@ func (s *ArchiveSuite) TestPostEntityClearsCanIngest(c *gc.C) {
 	})
 }
 
+func (s *ArchiveSuite) TestPostEntityWithIngestDoesNotClearCanIngest(c *gc.C) {
```
Comment (mhilton) [inline internal/v5/archive_test.go]: This test doesn't seem to tie up with the title

File: internal/v5/archive_test.go [hunk 2]
```diff
@@ -672,6 +672,39 @@ func (s *ArchiveSuite) TestPostEntityClearsCanIngest(c *gc.C) {
...
+	id := newResolvedURL("~charmers/precise/juju-gui-0", -1)
+	s.assertUploadCharm(c, "PUT", id, "wordpress", nil)
+	s.setPublic(c, id)
+
+	// Sanity check that can-ingest is false after the initial PUT.
```
Comment (mhilton) [inline internal/v5/archive_test.go:680]: s/false/true

**Comment by mhilton** [comment]
:+1:

**Comment by frankban** [comment]
:+1:

**Comment by rogpeppe** [comment]
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/charmstore-merge

**Pull request merged**